crates/sui-core/src/transaction_manager.rs

```diff
@@ -196,11 +199,7 @@ impl TransactionManager {
         for digest in ready_digests.iter() {
             // NOTE: failing and ignoring the certificate is fine, if it will be retried at a higher level.
             // Otherwise, this has to crash.
-            let cert = match self
-                .authority_store
-                .epoch_store()
-                .get_pending_certificate(digest)
-            {
+            let cert = match epoch_store.get_pending_certificate(digest) {
```

Comment:
The existing logic is incorrect with epoch changes, but I'm not sure if the new logic is correct either. Basically, I'm not sure if it is possible to have transactions from multiple epochs pending in transaction manager together. If it is possible, `digest` here may be from a later epoch. I think we can store pending certificates in transaction manager in memory to skip fetch from epoch store.

Either way, we can also turn the fetch below to panic on certificate not found.

In this case, suppose txn 1 from epoch 1 creates object A, that is the input to txn 2 from epoch 2. Both Commit call to

Reply:
Does https://github.com/MystenLabs/sui/pull/7134 resolve your concern?

Reply:
Discussed off thread. It is guaranteed that `AuthorityState` and `TransactionManager` are active for only 1 epoch at a time. We finish executing all transactions from an epoch that are included in all checkpoints, before moving to the next epoch. So if a certificate from the previous epoch still exists in TxManager, it does not make into any checkpoint and can be ignored. Later, we can clear TxManager at epoch boundary to ensure data consistencies.